src/Kiota.Builder/Extensions/OpenApiOperationExtensions.cs

```diff
@@ -10,27 +11,33 @@ public static class OpenApiOperationExtensions {
             "application/json",
             "text/plain"
         };
+        /// <summary>
+        /// cleans application/vnd.github.mercy-preview+json to application/json
+        /// </summary>
+        private static readonly Regex vendorSpecificCleanup = new(@"[^/]+\+", RegexOptions.Compiled);
         public static OpenApiSchema GetResponseSchema(this OpenApiOperation operation)
         {
             // Return Schema that represents all the possible success responses!
-            // For the moment assume 200s and application/json
             var schemas = operation.Responses.Where(r => successCodes.Contains(r.Key))
-                                .SelectMany(re => re.Value.Content)
-                                .Where(c => validMimeTypes.Contains(c.Key))
-                                .Select(co => co.Value.Schema)
+                                .SelectMany(re => re.Value.GetResponseSchemas())
                                 .Where(s => s is not null);
 
             return schemas.FirstOrDefault();
         }
-        public static OpenApiSchema GetResponseSchema(this OpenApiResponse response)
+        public static IEnumerable<OpenApiSchema> GetResponseSchemas(this OpenApiResponse response)
         {
-            // For the moment assume application/json
             var schemas = response.Content
-                                .Where(c => validMimeTypes.Contains(c.Key))
+                                .Where(c => !string.IsNullOrEmpty(c.Key))
+                                .Select(c => (Key: c.Key.Split(';', StringSplitOptions.RemoveEmptyEntries).FirstOrDefault(), c.Value))
+                                .Where(c => validMimeTypes.Contains(c.Key) || validMimeTypes.Contains(vendorSpecificCleanup.Replace(c.Key, string.Empty)))
                                 .Select(co => co.Value.Schema)
                                 .Where(s => s is not null);
 
-            return schemas.FirstOrDefault();
+            return schemas;
+        }
+        public static OpenApiSchema GetResponseSchema(this OpenApiResponse response)
+        {
+            return response.GetResponseSchemas().FirstOrDefault();
```

Comment:
Are we planning to use the schema list in its entirety later on? Seems like a waste to compute multiple schemas just to discard them later 🤔